src/id_token_provider.erl

```diff
@@ -96,10 +97,14 @@ refresh(Provider) ->
 
 refresh(Provider, WellKnownUri) ->
   {ok, KeysUrl} = id_token_jwks:get_jwks_uri(WellKnownUri),
-  {ok, NewKeys} = id_token_jwks:get_pub_keys(KeysUrl),
-  NewCacheEntry = NewKeys#{well_known_uri => WellKnownUri},
-  ets:insert(?ID_TOKEN_CACHE, {Provider, NewCacheEntry}),
-  NewKeys.
+  case id_token_jwks:get_pub_keys(KeysUrl) of
+    {ok, NewKeys} ->
+      NewCacheEntry = NewKeys#{well_known_uri => WellKnownUri},
+      ets:insert(?ID_TOKEN_CACHE, {Provider, NewCacheEntry}),
+      {ok, NewKeys};
+    {error, Reason} ->
```

Comment:
I was thinking, why not just return the old ones? Worse case it won't find the right key and then it would fail validation, but it will likely find it as, as you pointed out, the refresh happens more than what is actually needed

Reply:
Maybe just log an error here, so there's a trace of it happening

Reply:
sounds good

Reply:
done